File: MsgServerClientSdk/SdkMsgClientAndroid/sdkmsgclient/src/main/java/org/dync/teameeting/sdkmsgclient/msgs/MSClientManager.java
package org.dync.teameeting.sdkmsgclient.msgs;

import android.content.Context;

/**
 * Created by hp on 7/10/16.
 */
public class MSClientManager {
    /**
     *  init msg client
     *
     *  params:
     *  usrId: the userid
     *  token: the user token
     *  nName: the user nick name
     *
     *  return:
     *      0: init ok
     *      -101: one of params are null
     */
    public int initMsgClient(Context context, String usrId, String token, String nName) {
        System.out.println("MsgClient Android Version:v1.0.0");
        if (null == usrId || usrId.length()==0 || null == token || token.length()==0 || null == nName || nName.length()==0) return -101;
        MsgClient.getInstance().MCInit(context, usrId, token, nName);
        MsgClient.getInstance().MCRegisterMsgCb();
        return 0;
    }

    /**
     *  uninit msg client
     *
     *  params:
     *
     *  return:
     *      0: unin ok
     */
    public int uninMsgClient() {
        MsgClient.getInstance().MCUnRegisterMsgCb();
        MsgClient.getInstance().MCUnin();
        return 0;
    }

    /**
     *  add client delegate
     *
     *  params:
     *      cltDelegate: the client delegate for callback to set
     *
     *  return:
     *      void
     */
    public void addDelegate(MSClientDelegate cltDelegate) {
        MsgClient.getInstance().MCSetClientDelegate(cltDelegate);
    }

    /**
     *  remove client delegate
     *
     *  params:
     *      cltDelegate: the client delegate to delete
     *
     *  return:
     *      void
     */
    public void delDelegate(MSClientDelegate cltDelegate) {
        MsgClient.getInstance().MCSetClientDelegate(null);
    }

    /**
     *  connect to server
     *
     *  params:
     *      server: the msg server ip
     *      port: the msg server port
     *
     *  return:
     *      0: connect server ok
     *      -101: params is null
     */
    public int connToServer(String server, int port) {
        if (null == server || server.length()==0 || port <0) return -101;
        return MsgClient.getInstance().MCConnToServer(server, port);
    }

    /**
     *  get connection status
     *
     *  params:
     *
     *  return:
     *      int: the state of the connection to server
     */
    public int connStatus() {
        return MsgClient.getInstance().MCConnStatus();
    }

    /**
     *  set userid
     *
     *  params:
     *      userid: the userid to set
     *
     *  (but this func has not implement)
     *
     *  return:
     *      void
     *
     */
    public void setUserid(String strUserid) {
        //DO NOT IMPLEMENT
    }

    /**
     *  set user token
     *
     *  params:
     *      token: the user token
     *
     *  return:
     *      void
     */
    public void setToken(String strToken) {
        if (null == strToken || strToken.length()==0) return;
        MsgClient.getInstance().MCSetToken(strToken);
    }

    /**
     *  set user nick name
     *
     *  params:
     *      nName: the user nick name
     *
     *  return:
     *      void
     */
    public void setNickName(String strNname) { if (null == strNname || strNname.length()==0) return;
        MsgClient.getInstance().MCSetNickName(strNname); }

    /**
     *  set user icon url
     *
     *  params:
     *      url: the user icon url
     *
     *  return:
     *      void
     */
    public void setUIconUrl(String strUicon) { if (null == strUicon || strUicon.length()==0) return;
        MsgClient.getInstance().MCSetUIconUrl(strUicon); }

    /**
     *  set if need push
     *
     *  params:
     *      push: 1, enable push, 2, disable push
     *
     *  return:
     *      void
     */
    public void enablePush(int push) {
        MsgClient.getInstance().MCEnablePush(push);
    }

    /**
     *  if mute notifications
     *
     *  params:
     *      mute: 1, mute notification, 2, not mute notification
     *
     *  return:
     *      void
     */
    public void muteNotify(int mute) {
        MsgClient.getInstance().MCMuteNotify(mute);
    }
}

File: MsgServer/MsgServerClient/ClientCommon/MSClientManager.cpp
//
//  MSClientManager.m
//  SdkMsgClientIos
//
//  Created by hp on 6/29/16.
//  Copyright © 2016 Dync. All rights reserved.
//

#include "MSClientManager.h"
#include "MsgClient.h"


int MSClientManager::InitMsgClient(const std::string& usrId, const std::string& token, const std::string& nName)
{
    LI("MSClientManager Linux Version:v1.0.0");
    if (usrId.length()==0||token.length()==0||(nName.length()==0)) return -101;
    MsgClient::Instance().MCInit(usrId, token, nName);
    MsgClient::Instance().MCRegisterMsgCb(&MsgClient::Instance());
    return 0;
}

int MSClientManager::UninMsgClient()
{
    MsgClient::Instance().MCUnRegisterMsgCb(&MsgClient::Instance());
    MsgClient::Instance().MCUnin();
    return 0;
}

void MSClientManager::AddDelegate(const MSClientDelegate* cltDelegate)
{
    MsgClient::Instance().MCSetClientDelegate(cltDelegate);
}

void MSClientManager::DelDelegate(const MSClientDelegate* cltDelegate)
{
    MsgClient::Instance().MCSetClientDelegate(nullptr);
}

int MSClientManager::ConnToServer(const std::string& server, int port)
{
    if (server.length()==0 || port <0) return -101;
    MsgClient::Instance().MCConnToServer(server, port);
    return 0;
}

MCConnState MSClientManager::ConnStatus()
{
    MsgClient::Instance().MCConnStatus();
    return MCConnStateNOT_CONNECTED;
}

void MSClientManager::SetUserId(const std::string& userId)
{
   //DO NOT IMPLEMENT
}

void MSClientManager::SetToken(const std::string& token)
{
    if (token.length()==0) return;
    MsgClient::Instance().MCSetToken(token);
}

void MSClientManager::SetNickName(const std::string& nName)
{
    if (nName.length()==0) return;
    MsgClient::Instance().MCSetNickName(nName);
}

void MSClientManager::SetUIconUrl(const std::string& url)
{
    if (url.length()==0) return;
    MsgClient::Instance().MCSetUIconUrl(url);
}

void MSClientManager::EnablePush(int push)
{
    MsgClient::Instance().MCEnablePush(push);
}

void MSClientManager::MuteNotify(int mute)
{
    MsgClient::Instance().MCMuteNotify(mute);
}

